src/components/atoms/ToggleRP/ToggleRP.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react'
+
+import * as classNames from 'classnames'
+import { ClassNameProps } from 'interfaces/Props'
+import './ToggleRP.scss'
+
+interface ToggleProps extends ClassNameProps {
+  readonly children?: React.ReactNode
+  readonly on?: boolean
+  readonly labelledby?: string
+  readonly disabled?: boolean
+  readonly onClick?: () => void
+}
+
+interface ToggleState {
+  readonly on?: boolean
+}
+
+const callAll = (...fns: Array<() => void>) => (...args: any[]) => fns.forEach(fn => fn && fn(...args))
+
+export class ToggleRP extends React.Component<ToggleProps, ToggleState> {
+  readonly state = {
+    on: this.props.on ? true : false,
+  }
+
+  readonly toggle = () => {
+    this.setState({
+      on: !this.state.on,
+    })
+  }
+
+  readonly getToggleProps = (props: { readonly onClick?: any } = {}) => ({
+    onClick: callAll(props.onClick, this.toggle),
+  })
+
+  render(): React.ReactNode {
+    const { children } = this.props
+    return children ? (
+      children({
+        on: this.state.on,
+        toggle: this.toggle,
+        getToggleProps: this.getToggleProps,
+      })
+    ) : (
+      <button
```

Comment:
sometimes, it's better to use a checkbox for simulating a toggle. I leave the reference link 

https://www.w3schools.com/howto/howto_css_switch.asp

Reply:
We should have a component from a widget library for this. Do we have a component library?

Reply:
Is there a certain one we would like to use? And should I use it for this PR?